File: examples/tutorials/hello_user/python/src/undefined/hello_user_pygen.py
from __future__ import annotations
from typing import *
import examples.undefined.hello_user as pybind_hello_user_pygen
from metaclass import *
from bind_utils import *


class UserInfo(metaclass=OriginalMethodsMetaclass):
    """
    Structure to describe user."""
    @bind
    def __init__(self, user_name: str, user_age: int):
        pass


    @property
    @bind
    def age(self) -> int:
        """
        Age of user.
        """
        pass

    @property
    @bind
    def name(self) -> str:
        """
        Name of user.
        """
        pass

    @property
    @bind
    def want_a_drink(self) -> bool:
        """
        Some wishes of the user."""
        pass

    @want_a_drink.setter
    @bind
    def want_a_drink(self, value: bool):
        """
        Some wishes of the user."""
        pass
    


class Host(metaclass=OriginalMethodsMetaclass):
    """
    Host class."""
    @bind
    def __init__(self):
        pass


    
    @bind
    def hello(self, user: UserInfo) -> str:
        """
        Greeting function."""
        pass
    @bind
    def welcome(self, user: UserInfo) -> str:
        """
        Welcome function."""
        pass

File: examples/tutorials/hello_user/python/wrappers/undefined/hello_user_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/hello_user.hpp"

namespace py = pybind11;

void bindUserInfo(py::module& m) {
    using namespace hello_user;
    py::class_<hello_user::UserInfo> userinfo(m, "UserInfo");

    userinfo.def(py::init<const std::string &, unsigned int>(), py::arg("user_name"), py::arg("user_age"));

    userinfo.def_readonly("age", &hello_user::UserInfo::age);

    userinfo.def_readonly("name", &hello_user::UserInfo::name);

    userinfo.def_readwrite("want_a_drink", &hello_user::UserInfo::want_a_drink);
}
void bindHost(py::module& m) {
    using namespace hello_user;
    py::class_<hello_user::Host> host(m, "Host");

    host.def(py::init<>());
    host.def("hello", &hello_user::Host::hello, py::arg("user"));
    host.def("welcome", &hello_user::Host::welcome, py::arg("user"));
}